AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests.swift

```diff
@@ -0,0 +1,244 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import Amplify
+import AWSMobileClient
+import AWSAPICategoryPlugin
+@testable import AWSAPICategoryPluginTestCommon
+@testable import AmplifyTestCommon
+import AWSPluginsCore
+
+class GraphQLAuthDirectiveIntegrationTests: XCTestCase {
+
+    static let amplifyConfiguration = "GraphQLAuthDirectiveIntegrationTests-amplifyconfiguration"
+    static let awsconfiguration = "GraphQLAuthDirectiveIntegrationTests-awsconfiguration"
+    static let credentials = "GraphQLAuthDirectiveIntegrationTests-credentials"
+    static var user1: String!
+    static var user2: String!
+    static var password: String!
+
+    static override func setUp() {
+        do {
+
+            let credentials = try TestConfigHelper.retrieveCredentials(
+                forResource: GraphQLAuthDirectiveIntegrationTests.credentials)
+
+            guard let user1 = credentials["user1"],
+                let user2 = credentials["user2"],
+                let password = credentials["password"] else {
+                    XCTFail("Missing credentials.json data")
+                    return
+            }
+
+            GraphQLAuthDirectiveIntegrationTests.user1 = user1
+            GraphQLAuthDirectiveIntegrationTests.user2 = user2
+            GraphQLAuthDirectiveIntegrationTests.password = password
```

Comment:
ideally, user1 and user2 would have different passwords to avoid us thinking we are logging into user1 when we are really logging in with user2.

Reply:
updating